projects/generate_glitches/generate_glitches.py

```diff
@@ -0,0 +1,342 @@
+
+        except ValueError:
+            logging.info(f"Data not available for trigger at time: {time}")
+            continue
+
+    glitches = np.array(glitches)
+    snrs = np.array(snrs)
+    return glitches, snrs
+
+
+@typeo
+def main(
+    snr_thresh: float,
+    start: float,
+    stop: float,
+    window: float,
+    omicron_dir: str,
+    out_dir: str,
+    channel: str,
+    frame_type: str,
+    sample_rate: float = 4096,
+    H1_veto_file: Optional[str] = None,
+    L1_veto_file: Optional[str] = None,
+):
+
+    """Simulates a set of glitches for both
+        H1 and L1 that can be added to background
+
+    Arguments:
+
+    - snr_thresh: snr threshold above which to keep as glitch
+    - start: start gpstime
+    - stop: stop gpstime
+    - window: half window around trigger time to query data for
+    - sample_rate: sampling frequency
+    - out_dir: output directory to which signals will be written
+    - omicron_dir: base directory of omicron triggers
+            (see /home/ethan.marx/bbhnet/generate-glitch-dataset/omicron/)
+    - channel: channel name used to read data
+    - frame_type: frame type for data discovery w/ gwdatafind
+    - H1_veto_file: path to file containing vetoes for H1
+    - L1_veto_file: path to file containing vetoes for L1
+    """
+
+    # create logging file in model_dir
+    logging.basicConfig(
+        filename=f"{out_dir}/log.log",
+        format="%(message)s",
+        filemode="w",
+        level=logging.INFO,
+    )
+
+    # if passed, load in H1 vetoes and convert to gwpy SegmentList object
+    if H1_veto_file is not None:
+
+        logging.info("Applying vetoes to H1 times")
+        logging.info(f"H1 veto file: {H1_veto_file}")
+
+        # load in H1 vetoes
+        H1_vetoes = np.loadtxt(H1_veto_file)
+
+        # convert arrays to gwpy Segment objects
+        H1_vetoes = [Segment(seg[0], seg[1]) for seg in H1_vetoes]
+
+        # create SegmentList object
+        H1_vetoes = SegmentList(H1_vetoes).coalesce()
+    else:
+        H1_vetoes = None
+
+    if L1_veto_file is not None:
+        logging.info("Applying vetoes to L1 times")
+        logging.info(f"L1 veto file: {L1_veto_file}")
+
+        L1_vetoes = np.loadtxt(L1_veto_file)
+        L1_vetoes = [Segment(seg[0], seg[1]) for seg in L1_vetoes]
+        L1_vetoes = SegmentList(L1_vetoes).coalesce()
+    else:
+        L1_vetoes = None
+
+    # omicron triggers are split up by directories
+    # into segments of 10^5 seconds
+    # get paths for relevant directories
+    # based on start and stop gpstimes passed by user
+
+    # TODO: I *Think* there is a pyomicron package
+    # that can create omicron dags. Might be useful
+    # to generalize this script to use pyomicron to produce
+    # omicron triggers for arbitrary stretches of data,
+    # channels, parameters etc..
+
+    gps_day_start = str(start)[:5]
+    gps_day_end = str(stop)[:5]
+    all_gps_days = np.arange(int(gps_day_start), int(gps_day_end) + 1, 1)
```

Comment:
Minor point but would the same effect be achieved by ` // 10000` without having to go to strings and then back?